### PR TITLE
Properly handle repeating events

### PR DESCRIPTION
**Pull request by Naveen3Singh**
Fixes https://github.com/SimpleMobileTools/Simple-Calendar/issues/1708

Repeating events spanning midnight without glitches:

### CODE REVIEW

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt [hunk 1]
```diff
@@ -429,6 +433,11 @@ class MonthView(context: Context, attrs: AttributeSet, defStyle: Int) : View(con
         return event.startTS != event.endTS && Formatter.getDateTimeFromTS(event.endTS) == Formatter.getDateTimeFromTS(date.seconds()).withTimeAtStartOfDay()
     }
 
+    private fun findLastDay(event: Event) =
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt:436]: this is hard to read, type out the return value like (event: Event) : Day {... return.... please

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt [hunk 2]
```diff
@@ -113,16 +115,17 @@ class MonthView(context: Context, attrs: AttributeSet, defStyle: Int) : View(con
...
+        days.forEach { day ->
 
+            day.dayEvents.forEach { event ->
                 // make sure we properly handle events lasting multiple days and repeating ones
                 val lastEvent = allEvents.lastOrNull { it.id == event.id }
                 val daysCnt = getEventLastingDaysCount(event)
                 val validDayEvent = isDayValid(event, day.code)
-                if ((lastEvent == null || lastEvent.startDayIndex + daysCnt <= day.indexOnMonthView) && !validDayEvent) {
+
+                val lastDay = { findLastDay(event) ?: day }
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt:126]: do we need the brackets here?

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt:126]: ah, this is supposed to be a function? Make it a proper function then please, this is unreadable

Comment (Naveen3Singh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt:126]: Yes, it's a lambda

Without the lambda, the `findLastDay()` (iterates through the whole `days` array and events) will be executed needlessly in each loop even when there are no repeating events 

I created the lambda for readability only, the `findLastDay() :? day` term could be moved to the if statement itself avoiding the need for a lambda

How about local functions? Allowed?

Comment (tibbi) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt:126]: that sounds really bad, just add some casual function please

**Comment by tibbi** [comment]
yeah thanks, looks much better now

File: app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt (hunk 2 above)
Comment (Naveen3Singh) [reply app/src/main/kotlin/com/simplemobiletools/calendar/pro/views/MonthView.kt]: https://github.com/SimpleMobileTools/Simple-Calendar/pull/1722/commits/ac55c060d51f53723ae43fad7955d7cd2cfe6675

(the nullability check was unnecessary)

**Pull request merged**